msal/application.py

```diff
@@ -533,6 +551,12 @@ def _validate_ssh_cert_input_data(self, data):
                     "you must include a string parameter named 'key_id' "
                     "which identifies the key in the 'req_cnf' argument.")
 
+    def _get_new_correlation_id(self):
```

Comment:
@sangonzal Good job on noticing the underscore prefix naming convention for non-public methods!

Given that these 2 new helpers do not share any data with the `ClientApplication` class, I would suggest to put them as 2 highest level functions.